src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -10251,7 +10251,7 @@ namespace SurfaceGeometry {
                 WindowShadingControl(a).SetPoint == WindowShadingControl(b).SetPoint &&
```

Comment:
In the function above this `CheckWindowShadingControlSimilarForWindow`, in the outer loop `(auto theSurf :` should be `(auto &theSurf`. I don't know this for sure, but I'm afraid that omitting the `&` causes the loop induction variable to be a **copy** of the surface struct. Better add `&` just to make sure.